Ddf 3357

gordocanchola opened the pull request:
Please review this.
@clockard @djblue @vinamartin @stustison @oscaritoro @ricklarsen @andrewkfiedler 

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/313)
<!-- Reviewable:end -->

vinamartin commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java:
```diff
@@ -0,0 +1,364 @@
+
+    private SystemPropertyDetails getSystemPropertyDetails(String key, String title,
+            String description, List<String> options) {
+        String defaultValue = "";
```
I'm a little confused as to why defaultValue and value are separate variables when they end up getting the same value. I understand that in the constructor they are different, just a little confused here.

stustison commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java:
```diff
@@ -0,0 +1,364 @@
+/**
+ *
+ */
+public class SystemPropertiesAdmin implements SystemPropertiesAdminMBean {
```
This should be in its own module, not in admin core api.

stustison commented on platform/admin/core/admin-core-api/pom.xml:
```diff
@@ -66,6 +66,26 @@
+            <groupId>ddf.security.certificate</groupId>
+            <artifactId>security-certificate-generator</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
These dependencies should be placed in your own module, they can't be in admin-core-api. This is going to create a circular dependency to security that should not exist.

clockard replied:
Removed the certs part of the SystemPropertiesAdmin and these dependencies.

stustison commented on platform/admin/core/admin-core-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml:
```diff
@@ -35,6 +35,10 @@
 
+    <bean id="systemProperties" class="org.codice.ddf.ui.admin.api.impl.SystemPropertiesAdmin" destroy-method="shutdown" />
+
+    <!--<service ref="systemProperties" interface="org.codice.ddf.ui.admin.api.SystemPropertiesAdminMBean" />-->
```
This should be removed and dropped into its own module.

vinamartin commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java:
```diff
@@ -0,0 +1,364 @@
+            LOGGER.error("Error adding system key to keystore.", e);
+        } catch (Exception e) {
+            errors.add("Error communicating with KeystoreEditorMBean. " + e.getMessage());
+            LOGGER.error("JMX error updating certs", e);
```
Why aren't the errors and logger messages the same here, although they are above? 
Should they be the same or is there a reason why they're different here?

clockard replied:
This is OBE now that we have reworked the certs part of the mbean

stustison commented on platform/admin/ui/Gruntfile.js:
```diff
@@ -173,8 +173,8 @@ module.exports = function (grunt) {
 
     grunt.registerTask('build', ['bower-offline-install', 'sed', 'newer:less',
-        'newer:cssmin', 'newer:jshint']);
+        'newer:cssmin'/*, 'newer:jshint'*/]);
```
You need to put jshint back in, this is to restrict our JS to conform to our standards. If jshint isn't passing, you need to fix the JS.

clockard commented:
Reviewed 39 of 39 files at r1.
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 44 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30TIVe9OhxWu5i60SE)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L44)):</sup>
Probably could add some javadoc here. If not then remove the empty comment.
<!-- Reviewable comment -K30TIVe9OhxWu5i60SF:-495105096 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30VZ8QIJBXiMfrLXPC)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L126)):</sup>
Might consider moving this block into the configureMBean() method
<!-- Reviewable comment -K30VZ8QIJBXiMfrLXPD:-1106622154 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 225 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Ub2At-9QlhgMgjHy)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L225)):</sup>
This could be simplified by doing String property = System.getProperty(key,""); and then just pass the property as the value and default. No need for the if statement and extra variables then.
<!-- Reviewable comment -K30Ub2At-9QlhgMgjHz:-53238400 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 255 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30VfCnHnw3Lnz8UtVy)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L255)):</sup>
Should probably at least log something here
<!-- Reviewable comment -K30VfCnHnw3Lnz8UtVz:691511873 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 362 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30XcgbEpXJtZX_VBsb)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L362)):</sup>
Should this method unregister the mbean?
<!-- Reviewable comment -K30XcgbEpXJtZX_VBsc:2107186288 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminMBean.java, line 22 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30XtEvpXciqAUGouBr)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminMBean.java#L22)):</sup>
Since this is an interface we should probably add some javadocs to the methods
<!-- Reviewable comment -K30XtEvpXciqAUGouBs:-967276741 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Y6AgB71GD82tLyhK)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L40)):</sup>
Removed commented out line
<!-- Reviewable comment -K30Y6AgB71GD82tLyhL:980279501 -->

---

<sup>**[platform/admin/ui/Gruntfile.js, line 176 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Yt8_Q_w68-ncAA2E)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/Gruntfile.js#L176)):</sup>
Is this suppose to be commented out? If so just remove it.
<!-- Reviewable comment -K30Yt8_Q_w68-ncAA2F:1825221399 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/models/Installer.js, line 15 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30ZS3iR2_rJmcYDVnt)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/models/Installer.js#L15)):</sup>
I don't know that we want to use alerts in the installer. Consider replacing. 
<!-- Reviewable comment -K30ZS3jlHrG4TBgJsDL:-360996475 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/models/Installer.js, line 151 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Zkgymx6IB3mtZgWH)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/models/Installer.js#L151)):</sup>
Same comment as above.
<!-- Reviewable comment -K30Zkgymx6IB3mtZgWI:886361481 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 69 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30_hzz96hiXEvoev3y)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L69)):</sup>
removed commented out options
<!-- Reviewable comment -K30_hzz96hiXEvoev3z:1302104901 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 125 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30a10jPdT3TglmW6HO)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L125)):</sup>
Since we can't go back we should probably just have the next button.
<!-- Reviewable comment -K30a10jPdT3TglmW6HP:-1513965479 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 219 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30aVDmB0YC1blXPhcZ)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L219)):</sup>
I think we can remove this. See comment from above
<!-- Reviewable comment -K30aVDmB0YC1blXPhc_:318994697 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/InstallerMain.view.js, line 191 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30bBeWj4bS3eAFiZBj)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/InstallerMain.view.js#L191)):</sup>
Probably should revert this file change to remove it from the pr.
<!-- Reviewable comment -K30bBeWj4bS3eAFiZBk:1164001086 -->

---

<sup>**[platform/admin/ui/src/main/webapp/main.js, line 32 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30bowbnK74k0B_p0-l)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/main.js#L32)):</sup>
should switch these back to the minified versions
<!-- Reviewable comment -K30bowbnK74k0B_p0-m:-1491639546 -->

---

<sup>**[platform/admin/ui/src/main/webapp/templates/installer/configuration.handlebars, line 7 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30c4Qctv5vi2IPE9wD)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/templates/installer/configuration.handlebars#L7)):</sup>
Move the copyright to the top
<!-- Reviewable comment -K30c4Qctv5vi2IPE9wE:1603585331 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

vinamartin commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java:
```diff
@@ -0,0 +1,364 @@
+        return certMbeanProxy;
+    }
+
+    public void shutdown() {
```
Is this used?

vinamartin commented on platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java:
```diff
@@ -0,0 +1,241 @@
+    public void testReadSystemProperties() {
+        SystemPropertiesAdmin spa = new SystemPropertiesAdmin();
+        List<SystemPropertyDetails> details = spa.readSystemProperties();
+        assertThat(details.size(), is(9));
```
Should we rename this or adjust the assertion to really test that the system properties are being read correctly? 
If I understand this correctly it's just asserting that 9 properties were read

clockard replied:
Updated assertions to verify all the property values.

ricklarsen commented:
Reviewed 39 of 39 files at r1.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

vinamartin commented on platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java:
```diff
@@ -0,0 +1,241 @@
+    }
+
+    @Test
+    public void testWriteSystemPropertie() throws Exception {
```
Missing an s after `Propertie`

stustison replied:
typo

clockard replied:
fixed

ricklarsen commented:
:+1:

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/AnonClaims.view.js:
```diff
@@ -81,8 +81,6 @@ define([
                 this.valObj.get("claims").availableClaims.push("Add Custom Attribute...");
             }
-
-            Backbone.ModelBinder.SetOptions({modelSetOptions: {validate: true}});
```
Why did we remove this?

clockard replied:
This sets the property globally which we don't want. It was a mistake I made when I added the AnonClaims and it was causing problems with the configuration page.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js:
```diff
@@ -0,0 +1,119 @@
+        },
+        saveCerts: function () {
+            var view = this;
+            this.listenTo(this.model, 'invalid', function (model, errors) {
```
Marionette has a modelEvents object that can be added to the view to trigger model events for you instead of dropping it here.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js:
```diff
@@ -19,28 +19,242 @@ define([
+    ich.addTemplate('configurationItemTemplate', configurationItemTemplate);
 
 
+/*
```
These models should not be in the view. They should be moved to js/models/installer

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js:
```diff
@@ -19,28 +19,242 @@ define([
+                        hostChange = false;
+                    }
+                } else if(model.get('title') === 'HTTPS Port') {
+                    layout.navigationModel.set('redirectUrl','https://localhost:'+model.get('value')+'/admin/index.html');
```
You cannot use a fully qualified URL in the web app, this is going to break in production.

vinamartin commented on platform/security/certificate/security-certificate-keystoreeditor/src/test/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditorTest.java:
```diff
@@ -296,6 +323,67 @@ public void testAddKeyLocal() throws KeystoreEditor.KeystoreEditorException, IOE
+        byte[] crtBytes = IOUtils.toByteArray(fileInputStream);
+        byte[] encodedBytes = Base64.encode(crtBytes);
+        IOUtils.closeQuietly(fileInputStream);
+        keystoreEditor.addAllKeystoreEntries("changeit", "changeit", new String(encodedBytes), null,
```
Should be a system property for the password that you can use.

stustison commented on platform/admin/ui/src/main/webapp/main.js:
```diff
@@ -29,8 +29,8 @@
             // TODO test combining
-            modelbinder: 'lib/backbone.modelbinder/Backbone.ModelBinder.min',
-            collectionbinder: 'lib/backbone.modelbinder/Backbone.CollectionBinder.min',
+            modelbinder: 'lib/backbone.modelbinder/Backbone.ModelBinder',
```
Put the minified versions back in.

stustison commented on platform/admin/ui/src/main/webapp/templates/installer/navigationButtons.handlebars:
```diff
@@ -43,9 +43,11 @@
         {{else}}
-            <a id="finishStep" class="next finish" href="#">Finish</a>
+            <a id="restartLaterStep" class="next restart-later" href="#">Restart Later</a>
+            <a id="restartNowStep" class="next restart-now" href="#">Restart Now</a>
```
The Restart Later/Now buttons should only show if they actually changed something. If someone just hit next all the way through the installer, it should still just say Finish.

stustison commented on platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java:
```diff
@@ -270,13 +352,16 @@ private synchronized void addToStore(String alias, String keyPassword, String st
-                    } else {
-                        Certificate certificate = jks.getCertificate(alias);
-                        store.setCertificateEntry(alias, certificate);
+                    if (alias.equals(jksAlias)) {
```
This is going to break uploading a jks. We should be basically taking every single key in the JKS, but checking that any specified alias exists in it.

clockard replied:
This check was added because otherwise if you had more than one entry in passed in jks file it would try to add them all with the same alias. This has been fixed to instead just add all the entries with the same alias as found in the passed in jks file.

pklinef commented on platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java:
```diff
@@ -0,0 +1,364 @@
+    }
+
+    @Override
+    public List<String> setSystemCerts(Map<String, Object> certProps) {
```
This is not related system properties and should be in a different or new mbean.

stustison replied:
It seems like we could just hit the jolokia KeystoreEditor endpoints directly instead of creating your own custom mbean that just calls the KeystoreEditor stuff directly. Then there is no hard dependency between these two things.

clockard replied:
The KeystoreEditor and CertificateGenerator have been updated to contain the needed functionality to remove the need for a new mbean. All cert related stuff has been removed from the SystemPropertiesAdmin.

stustison commented on platform/admin/ui/src/main/webapp/js/models/CertsModel.js:
```diff
@@ -0,0 +1,112 @@
+            truststoreFileName: undefined,
+            truststorePass: undefined,
+            certErrors: [],
+            devMode: Backbone.history.location.search.indexOf('dev=true') > -1 ? true : false
```
We don't actually use backbone history, so what is this doing?

clockard replied:
Just trying to get the info the "backbone" way. Changed it to just use the location for clarity.

stustison commented on platform/admin/ui/src/main/webapp/js/models/CertsModel.js:
```diff
@@ -0,0 +1,112 @@
+
+            return validation.length > 0 ? validation : undefined;
+        },
+        sync: function () {
```
It looks like this sync was more/less copied from Service.js (but it is still using the deprecated .done() function that was just removed). This might be the time to break that sync out of Service.js and have a Jolokia model that all these other models can then extend.

clockard replied:
With the other updates the sync function changed to incorporate additional logic for choosing which mbean to use and setting up the arguments accordingly.

stustison commented on platform/admin/ui/src/main/webapp/js/models/FileHelper.js:
```diff
@@ -0,0 +1,62 @@
+            }, this);
+            reader.readAsDataURL(this.file);
+        },
+        toJSON: function () {
```
Why did we write our own toJSON function? If you need to exclude something from the resulting json, there are options that you can pass to a backbone association model to include only specific attributes.

clockard replied:
This was actually a helper js file copied from the certs editor admin page.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js:
```diff
@@ -0,0 +1,119 @@
+        },
+        showCertErrors: function () {
+                this.render();
+                this.$('.cert-errors').show();
```
Rather than using jquery to do this, you could hook up a modelEvent to re-render the view if the model changes. Then in your template you could show this section if certErrors.length > 0. Then this entire function goes away.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js:
```diff
@@ -19,28 +19,242 @@ define([
+                    _.each(error, function(errorItem){
+                       if (m.get('key') === errorItem.id) {
+                        layout.$("[name='" + m.get('key') + "Error']").show().html(errorItem.message);
+                       }
```
This all looks like stuff that should be in the template.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js:
```diff
@@ -19,28 +19,242 @@ define([
+
+                this.model.each(function(m){
+                    if (m.get('key') === model.get('key')) {
+                        layout.$("[name='" + m.get('key') + "Error']").hide();
```
Same here, template?

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js:
```diff
@@ -0,0 +1,119 @@
+                    if (errorItem.name) {
+                        view.$('[name=' + errorItem.name + ']').addClass('error-border');
+                    } else if (errorItem.id) {
+                        view.$('[id=' + errorItem.id + ']').addClass('error-border');
```
This all looks like stuff that should be in the template.

gordocanchola replied:
These values are validated together as one. Didn't find a clean way to move it to the template without adding a bunch of things to the model.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js:
```diff
@@ -19,28 +19,242 @@ define([
+                    propertySave.done(function(){
+                        if(hostChange) {
+
+                            var certSave = layout.certificates.currentView.saveCerts();
```
If you created the certificate model that was then passed to this view, you could simply call save() on that model and have a listener in the certificates view to listen to that call rather than trying to dig down into the layout to call the view directly. Doing this in backbone is not using their api correctly and could break in a future release of backbone.

stustison commented on platform/admin/ui/src/main/webapp/js/views/installer/Finish.view.js:
```diff
@@ -31,17 +31,23 @@ define([
             //this is your hook to perform any teardown that must be done before going to the previous step
             this.navigationModel.previousStep();
+        },
+        showLoading: function(){
```
What is this function doing? What causes the overlay to toggle off and the main content to come back. Also, these things look like they should trigger off of some sort of model change event and simply re-render the view.

gordocanchola replied:
At the point where this is displayed, there is no toggle off. The system is restarting.

stustison commented on platform/admin/ui/src/main/webapp/templates/installer/certificate.handlebars:
```diff
@@ -0,0 +1,88 @@
+                        <span>Select file to upload</span>
+                        <input id="inputFile" type="file" name="files[]">
+                    </span>
+                OR drag file to this window.
```
Is it actually to this window or to this "box" same for the one below.

stustison commented on platform/admin/ui/src/main/webapp/templates/installer/configuration.handlebars:
```diff
@@ -11,16 +17,3 @@
-            to change hostname, ports and certificates. A system shutdown is required in order to make these changes.<br/><br/>
-
-            Global system properties can be configured in the INSTALL_HOME/etc/system.properties file and certificates for the host can be added to the systemKeystore/systemTruststore in INSTALL_HOME/etc/keystore/<br/>
-
```
I think we still need all this text in here still to let admins know that they can use the system.properties file to make additional changes.

gordocanchola replied:
Added the missing text

stustison commented on platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditorMBean.java:
```diff
@@ -30,12 +30,86 @@
     List<Map<String, Object>> getTruststore();
 
-    void addPrivateKey(String alias, String keyPassword, String storePassword, String data, String type, String fileName)
+    void reInitializeKeystores();
```
I don't think this is needed. This looks like a bug in the current implementation of this class to me. We should just always be reading from the keystores whenever an action is performed, rather than caching them.

clockard replied:
Removed this function.

oscaritoro commented:
Technically, shouldn't this have been DDF-1525? Got a little confused there between the DDF and DIB tickets.

oscaritoro commented:
Reviewed 6 of 39 files at r1.
Review status: all files reviewed at latest revision, 43 unresolved discussions.

---

<sup>**[platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java, line 87 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o9-r1-87)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java#L87)):</sup>
Yes, this is a somewhat finicky test. Add or remove properties and it's broken... Maybe it's enough, maybe it's too much for what you really mean here (in which case maybe better to go with just an "isnot(0)"), or maybe it's not enough and you need to check deeper with something like "assertTrue(details.contains(something));") I don't know enough or who should make that decision... At a minimum I do see you using that literal 9 in many places, might be better to create a constant so that if it ever changes you only change it in one place.
<!-- Reviewable comment -K31oJ7uCuSNx9NjvvQO:-189515395 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

oscaritoro commented:
Reviewed 9 of 39 files at r1.
Review status: all files reviewed at latest revision, 43 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

oscaritoro commented:
Reviewed 2 of 39 files at r1.
Review status: all files reviewed at latest revision, 43 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

gordocanchola commented:
Review status: all files reviewed at latest revision, 42 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 44 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30TIVe9OhxWu5i60SE)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L44)):</sup>
Removed it
<!-- Reviewable comment -K3HkRDAV31dNyGaRJiu:1668562896 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 46 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o8-r1-46)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L46)):</sup>
Removed the certs from this MBean so it now fits better admin core api.
<!-- Reviewable comment -K3HkVz9l-kvTKN7ABEM:-1889951101 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30VZ8QIJBXiMfrLXPC)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L126)):</sup>
Moved to configureMBean
<!-- Reviewable comment -K3HlYsAvhix_WoE3zCo:1514697674 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 223 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o8-r1-223)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L223)):</sup>
I was doing something different with each that was no longer needed. It was just overlooked. Sorry for the confusion.
<!-- Reviewable comment -K30zAYuRGicCk_-6Una:-1418463234 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 225 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Ub2At-9QlhgMgjHy)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L225)):</sup>
Changed to provide the default value;
<!-- Reviewable comment -K3Hlu9dCKUjqAA_JeE9:403204459 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 255 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30VfCnHnw3Lnz8UtVy)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L255)):</sup>
Added the error message
<!-- Reviewable comment -K3HlzQQHA0bqhwRIZF1:-2105240933 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 362 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30XcgbEpXJtZX_VBsb)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L362)):</sup>
Modified to unregister the MBean
<!-- Reviewable comment -K3HmuiRShrkKh4Wl_zb:1484019613 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminMBean.java, line 22 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30XtEvpXciqAUGouBr)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminMBean.java#L22)):</sup>
Added Javadocs
<!-- Reviewable comment -K3Hn1Medo-EatmCPTYA:-611212542 -->

---

<sup>**[platform/admin/core/admin-core-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Y6AgB71GD82tLyhK)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L40)):</sup>
Removed commented out line
<!-- Reviewable comment -K3Hn8Ox_aOYkSRe-r2y:980279501 -->

---

<sup>**[platform/admin/ui/Gruntfile.js, line 176 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORa-r1-176)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/Gruntfile.js#L176)):</sup>
Put it back in. That change wasn't supposed to be committed. 
<!-- Reviewable comment -K31-JterXFMdCVqC9J9:1766191553 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/models/Installer.js, line 15 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30ZS3iR2_rJmcYDVnt)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/models/Installer.js#L15)):</sup>
This is the best option for this situation
<!-- Reviewable comment -K3HZpE93ULxboixF1mb:-1392527190 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/models/Installer.js, line 151 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30Zkgymx6IB3mtZgWH)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/models/Installer.js#L151)):</sup>
This is the best option for this situation
<!-- Reviewable comment -K3HZvwzCcz67YfTNAQY:-1392527190 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js, line 41 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7oC-r1-41)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js#L41)):</sup>
Changed to use modelEvents.
<!-- Reviewable comment -K3HJvygdHPDonqZoe_2:1330767897 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js, line 49 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7oC-r1-49)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js#L49)):</sup>
These values are validated together. Didn't find a clean way to move it to the template without adding a bunch of things to the model.
<!-- Reviewable comment -K3HXAKtc2lb0M3XiX9L:120465710 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js, line 72 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7oC-r1-72)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Certificate.view.js#L72)):</sup>
Removed the function.
<!-- Reviewable comment -K3HKPnVwG6pL1fa-0z6:-966134110 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 35 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORd-r1-35)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L35)):</sup>
Models were removed from the view and put into a model file
<!-- Reviewable comment -K3HKYauHv38GJb3edYD:-892684812 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 69 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30_hzz96hiXEvoev3y)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L69)):</sup>
Removed commented out options
<!-- Reviewable comment -K3HKtRDjvJTrpb7bajo:597687077 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 125 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30a10jPdT3TglmW6HO)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L125)):</sup>
Removed the previous functionality
<!-- Reviewable comment -K3H_CLolRMehqZXijlK:-1854076266 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 139 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORd-r1-139)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L139)):</sup>
Moved to template
<!-- Reviewable comment -K3HM9Q3aYRBHB2g5i8i:637887501 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 150 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORd-r1-150)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L150)):</sup>
Moved to template
<!-- Reviewable comment -K3HMBLfo3dSLgVgG8Zy:637887501 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 171 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORd-r1-171)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L171)):</sup>
Changed to a timed redirect. After the timeout if the system is not up the user will get a 404 and will have to manually refresh.
<!-- Reviewable comment -K3HWkTkEBE-zzEvvFWz:1419764356 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 183 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8Jld-aId-ZlJcORd-r1-183)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L183)):</sup>
Created the certificate model and using it directly.
<!-- Reviewable comment -K3HMFPrVRH-lPUeu1c1:2049471604 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js, line 219 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30aVDmB0YC1blXPhcZ)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Configuration.view.js#L219)):</sup>
Removed the previous functionality
<!-- Reviewable comment -K3H_JAUmirS06jXffe7:-1854076266 -->

---

<sup>**[platform/admin/ui/src/main/webapp/js/views/installer/Finish.view.js, line 48 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2sT_GrIyfOpfxaSGfZ-r1-48)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/js/views/installer/Finish.view.js#L48)):</sup>
At the point where this is displayed, there is no toggle off. The system is restarting.
<!-- Reviewable comment -K3HTtpyggy8k8oc7BMn:-2101487284 -->

---

<sup>**[platform/admin/ui/src/main/webapp/main.js, line 32 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlgNEIQqzB0DPo4-r1-32)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/main.js#L32)):</sup>
Done.
<!-- Reviewable comment -K3HMdp7Ux04a-OYvTCS:-499146009 -->

---

<sup>**[platform/admin/ui/src/main/webapp/main.js, line 32 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30bowbnK74k0B_p0-l)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/main.js#L32)):</sup>
Done.
<!-- Reviewable comment -K3HMeGkFHbrnPePGiyU:-499146009 -->

---

<sup>**[platform/admin/ui/src/main/webapp/templates/installer/certificate.handlebars, line 23 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7oD-r1-23)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/templates/installer/certificate.handlebars#L23)):</sup>
Changed to box
<!-- Reviewable comment -K3HMfdnssS4ZiFtv3M4:1833101815 -->

---

<sup>**[platform/admin/ui/src/main/webapp/templates/installer/configuration.handlebars, line 7 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K30c4Qctv5vi2IPE9wD)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/templates/installer/configuration.handlebars#L7)):</sup>
Moved to top
<!-- Reviewable comment -K3HMnMzt4LKi_EdxXpt:1566931426 -->

---

<sup>**[platform/admin/ui/src/main/webapp/templates/installer/navigationButtons.handlebars, line 50 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o--r1-50)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/ui/src/main/webapp/templates/installer/navigationButtons.handlebars#L50)):</sup>
Changed to only display the restart options if a change was detected.
<!-- Reviewable comment -K3H_Xl4CUaxhv511PAO:-867042809 -->

---

<sup>**[platform/security/certificate/security-certificate-keystoreeditor/src/test/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditorTest.java, line 335 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o5-r1-335)** ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/security/certificate/security-certificate-keystoreeditor/src/test/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditorTest.java#L335)):</sup>
Refactored to use a variable for the password.
<!-- Reviewable comment -K3HVSbnzwWU9-j0_FkS:1357079948 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)
<!-- Sent from Reviewable.io -->

clockard commented:
Reviewed 31 of 32 files at r2.
Review status: all files reviewed at latest revision, 38 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

oscaritoro commented:
Review status: all files reviewed at latest revision, 37 unresolved discussions.

---

_[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 190 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/313#-K3LkRFuHzm9iOpstVjP:-K3LkRFuHzm9iOpstVjQ:1923596256) ([raw file](https://github.com/codice/ddf/blob/80c964a1d197983368f7792cd116a6b033b49174/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L190)):_
Since you created this class/constructor do you still need to have that default value if it's the same?  maybe add another constructor?

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

clockard commented:
LGTM :+1:

vinamartin commented:
Looks good to me :+1: , though I see there's still some lingering comments from stustison that weren't addressed.

gordocanchola commented:
Review status: 39 of 45 files reviewed at latest revision, 34 unresolved discussions.

---

_[platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java, line 190 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/313#-K3LkRFuHzm9iOpstVjP:-K3Lv_DO0JJ6JHk3BnzT:253534921) ([raw file](https://github.com/codice/ddf/blob/80c964a1d197983368f7792cd116a6b033b49174/platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java#L190)):_
Removed the default value from the constructor and removed the setter for it also.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

gordocanchola commented:
Review status: 38 of 45 files reviewed at latest revision, 29 unresolved discussions.

---

_[platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java, line 87 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o9-r1-87:-K3PwZpoGG-9Mi9aljLo:1616890231) ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java#L87)):_
Added the checks for specific property values.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

oscaritoro commented:
Reviewed 8 of 41 files at r1, 32 of 32 files at r2, 6 of 6 files at r3.
Review status: 44 of 45 files reviewed at latest revision, 29 unresolved discussions.

---

_[platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java, line 87 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o9-r1-87:-K3QTDLXrucvNlobtUig:951043984) ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java#L87)):_
How about re-adding the ' assertThat(details.size(), is(X));' (along with verifying the values) to the tests (except testWriteSystemPropertiesNullProps() since that already has both?

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

djblue commented on platform/admin/ui/src/main/webapp/js/models/installer/CertsModel.js:
```diff
@@ -0,0 +1,132 @@
+            return validation.length > 0 ? validation : undefined;
+        },
+        sync: function () {
+            var deferred = $.Deferred(),
```
Are we using this deferred for anything?

gordocanchola replied:
Nope, not using it. Removed it.

djblue commented on platform/admin/ui/src/main/webapp/js/models/installer/CertsModel.js:
```diff
@@ -0,0 +1,132 @@
+            }
+
+
+            return validation.length > 0 ? validation : undefined;
```
Maybe for consistency with other validation functions, we could do:

```javascript
if (validation.length > 0) {
  return validation;
}
```

gordocanchola replied:
Changed match other validate function returns

djblue commented on platform/admin/ui/src/main/webapp/js/models/installer/Configuration.js:
```diff
@@ -0,0 +1,96 @@
+            return response.value;
+        },
+
+        save: function() {
```
This function is dope!

gordocanchola commented:
Review status: 44 of 45 files reviewed at latest revision, 31 unresolved discussions.

---

_[platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java, line 87 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/313#-K2J8JlhMvfT4lQAz7o9-r1-87:-K3RNDx-k75-sZRppLiy:-201920559) ([raw file](https://github.com/codice/ddf/blob/21d248d2e4a921bedb4b2106293434e59222bf1b/platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java#L87)):_
Added the size test.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

oscaritoro commented:
Reviewed 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 31 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313)_
<!-- Sent from Reviewable.io -->

oscaritoro commented:
lgtm :+1: 

---

Review status: all files reviewed at latest revision, 31 unresolved discussions.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/313#-:-K3RkitmA9Ia4bcHVcgZ:865466143)_
<!-- Sent from Reviewable.io -->
